cluster/cluster.go

```diff
@@ -126,28 +126,32 @@ func (c *clusterConnection) Do(ctx context.Context, req driver.Request) (driver.
 		timeout = c.defaultTimeout
 	}
 
-	serverCount := len(c.servers)
```

Comment:
here it is the race condition because `c.servers` can be written from the other place like `UpdateEndpoints`